### ISSUE TITLE
FEATURE: Remove out of scope domains from Alerts when the In Scope Button is pressed

### ISSUE THREAD

**Issue by globeone** [zaproxy/zaproxy#5869]
**Description**
As an IT Security Scan validator running 100s of security scans per week reports need to be free from any 'noise' in the form of out of scope domains. 
The current function removes the out-of-scope domains **Visually** but not from the report under the menu Reports... Generate HTML Report.

![image](https://user-images.githubusercontent.com/10052587/74825166-c13c7f80-5309-11ea-9703-58208bb46335.png)
![image](https://user-images.githubusercontent.com/10052587/74825809-d534b100-530a-11ea-992a-7f4f5468c4be.png)
![image](https://user-images.githubusercontent.com/10052587/74825186-c7caf700-5309-11ea-9afe-884238d20ccf.png)
![image](https://user-images.githubusercontent.com/10052587/74825834-e1207300-530a-11ea-8b6e-395b96204834.png)

**Describe the solution you'd like**
It would be helpful that if the Only In Scope button is pressed that the Report generated is also only in scope.  
![image](https://user-images.githubusercontent.com/10052587/74825489-4fb10100-530a-11ea-8236-f34f16e63bc0.png)

![image](https://user-images.githubusercontent.com/10052587/74825497-55a6e200-530a-11ea-8467-879d34fd143f.png)

**Describe alternatives you've considered**
At the moment I have to go through each alert and delete the huge swatches of mozilla.com, doubleclick.net, analytics.google.com etc.. before running a report which is very time-consuming.

Thanks.

**Comment by kingthorin** [comment]
The reporting suggestion makes sense. I think the Export Report add-on already has something along those lines.

> mozilla.com, doubleclick.net, analytics.google.com etc.. 

Sounds like you need to Enable (or add) some Global Exclude entries.
https://www.zaproxy.org/docs/desktop/ui/dialogs/options/globalexcludeurl/

**Comment by thc202** [comment]
Also, there's an option to not passive scan messages that are not in scope: https://www.zaproxy.org/docs/desktop/ui/dialogs/options/pscanner/

**Comment by globeone** [comment]
Thanks Mozilla! 
![image](https://user-images.githubusercontent.com/10052587/74826266-92270d80-530b-11ea-8ca8-307068f071da.png)

Thanks Google!
![image](https://user-images.githubusercontent.com/10052587/74826343-b8e54400-530b-11ea-863b-3e223e4ae1fa.png)

This is what's left after Deleting all the 'noise' from the 
![image](https://user-images.githubusercontent.com/10052587/74826508-019cfd00-530c-11ea-8a12-5caf1ce42915.png)

**Comment by globeone** [comment]
> Also, there's an option to not passive scan messages that are not in scope: https://www.zaproxy.org/docs/desktop/ui/dialogs/options/pscanner/

This doesn't work as the alerts are added to the scan as soon as you start the HUD.

**Comment by thc202** [comment]
That can be enabled before starting the HUD.

**Comment by globeone** [comment]
I'll try the setting again and see what happens. 

The point of this ticket is more useability that the 
![image](https://user-images.githubusercontent.com/10052587/74830345-36f91900-5313-11ea-8989-a3a061cd71db.png)
gives the impression that things will be kept in scope.

**Comment by globeone** [comment]
Sometimes it's good to see the out-of-scope alerts visually. 
For example, It's good to know what advertising tags are being used and which advertisers are not sending safe tags, I just don't want to report on them to the developer who can't do anything about it. The adverting tags come from different vendors, all with different responsible disclosure policies.
Using the option to turn off the passive scanner out of scope would mean running the full active and passive scans twice whereas pushing the already existing In Scope || Out of Scope button  
![image](https://user-images.githubusercontent.com/10052587/74831205-116d0f00-5315-11ea-8e6b-bd026dfe3692.png)
and running the report would be theoretically, more efficient.

**Comment by kingthorin** [comment]
I think it pretty clearly refers to display not anything else, but that’s just me.


Global Excludes give you control of that 3rd party cruft across all ZAP use, without having to define Context(s) [and therefore Scope].

**Comment by globeone** [comment]
Ok, I'll try the as-designed method with the Global Exclude and see how it goes. Thanks everyone for the location of the setting and the other 'insights' into the thinking behind this design descision.

**Comment by kingthorin** [comment]
I’m going to reopen until I can check the Export Report addon, as I said earlier it does make sense to have an ‘In-scope’ reporting option.

**Comment by ricekot** [comment]
Now that the new reports add-on allows selecting the context for reports, can this issue be closed?

**Comment by kingthorin** [comment]
I think so 😉

**Comment by thc202** [comment]
This feature request is being tracked in #6483 anyway.
